Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6502,6 +6517,26 @@ void HTMLMediaElement::remoteHasAvailabilityCallbacksChanged()
     mediaSession().setHasPlaybackTargetAvailabilityListeners(hasListeners);
     scheduleUpdateMediaState();
 }
+
+bool HTMLMediaElement::hasWirelessPlaybackTargetAlternative() const
+{
+    for (auto& source : childrenOfType<HTMLSourceElement>(*this)) {
+        String mimeType = source.attributeWithoutSynchronization(typeAttr);
+        ContentType contentType(mimeType);
+
+        if (contentType.containerType() == "application/vnd.apple.mpegurl"_s || contentType.containerType() == "application/x-mpegURL"_s)
```

Comment:
AppleTV supports other containers as well, why only HLS?

Reply:
there's no MSE/MMS on ATV. But the rule is to enable MSE/MMS if we have an HLS alternative so that AirPlay works with MSE

Reply:
> AppleTV supports other containers as well, why only HLS?

Oh I see what you mean; I guess I could do any alternative source not containing a MSE URL.